app/ircBot.js

```diff
@@ -0,0 +1,63 @@
+/*
+*  IrcBot.js handles all the commands being passed to the bot.
+*  Redirects to require library when a particular command is invoked.
+*/
+var irc = require('irc');
+var mysql = require('mysql');
+var botLib = require('./lib/bot');
+
+module.exports = function(config) {
+
+  //Create a irc client
+  var bot = new irc.Client(config.irc.server, config.irc.name, {
+    channels: config.irc.channels
+  });
+
+  /*
+  * On joining the channel, the bot checks for logged in user details.
+  * If user does not exist, then the bot adds the user to the database.
+  */
+  bot.addListener('join', function(channel, nick, message) {
+    if(nick != config.irc.name) {
+      var newUser = botLib.checkUser(nick);
+      return bot.say(channel, newUser);
+    }
+  });
+
+  /*
+  * Bot listens for commands beaing passed to it.
+  * Whenever a match is found, bot redirects to BotLib file.
+  */
+  bot.addListener('message', function(from, to, message) {
+    var cmdString = message.split(/@bot/)[1].trim();
```

Comment:
put validation and throw error if some wrong command send. 
This one fails with undefined doesnt have length if i just send a command `@bot`